manual/source/topic/pool.rst

```diff
@@ -170,3 +170,39 @@ Pool introspection
         at the address, use :c:func:`mps_addr_fmt`. If you only care
         whether the address belongs to a particular :term:`arena`, use
         :c:func:`mps_arena_has_addr`.
+
+
+.. c:function:: mps_res_t mps_pool_walk(mps_pool_t pool, mps_area_scan_t scan_area, void *closure)
+
+    Visit all :term:`formatted objects` in a :term:`pool`. The pool
+    must be :term:`automatically managed <automatic memory
+    management>`. The pool's :term:`arena` must be in the
+    :term:`parked state`.
+
+    :c:data:`pool` is the pool whose formatted objects are visited.
+
+    :c:data:`scan_area` is an area scanning function. See
+    :ref:`topic-scanning-area`.
+
+    :c:data:`closure` is an arbitrary pointer that will be passed to
+    :c:data:`scan_area`.
+
+    The scanning function is called multiple times with disjoint areas
```

Comment:
m: clarity - although the documentation is clear in explaining what the scanner must do, it's less clear what the scanner might be expected or permitted to do in common applications.